### PR TITLE
Add contributors and PR counts to release notes

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Add to the release notes:

- List of contributors, in order of their first contribution to this release
- Number of PRs merged

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

File: tools/CompileReleaseNotes.py [hunk 1]
```diff
@@ -129,6 +131,10 @@ def compile_release_notes(merged_prs: List[PullRequest]) -> str:
                (PULL_REQUEST_GROUPS.index(pr.group), merged_prs.index(pr))),
         key=lambda pr: pr.group)
 
+    # Get set of distinct PR authors. Maintain the order by using a dict, which
+    # is guaranteed to maintain the insertion order since Py 3.7
```
Comment (nilsdeppe) [inline tools/CompileReleaseNotes.py:135]: is there any chance the release env could use an older Python version and so we should do a sort just in case?

**Review by nilsdeppe — APPROVED** [review]

File: tools/CompileReleaseNotes.py (hunk 1 above)
Comment (nilsvu) [reply tools/CompileReleaseNotes.py:135]: The unit tests should fail if the ordering is wrong, I think that suffices.

**Review by nikwit — APPROVED** [review]
looks good to me. Out of interest, I understand these function annotations like in
```python
def get_merged_pull_requests(repo: git.Repo, from_rev: Revision,
                             to_rev: Revision) -> List[PullRequest]:
```

are in general ignored by python but used by static type checkers. What are we using here?

**Comment by nilsvu** [comment]
@nikwit Yes that's right. We currently don't run a Python linter on CI if that's what you're asking (https://github.com/sxs-collaboration/spectre/issues/1104).

**Pull request merged**